### PR TITLE
Runner changes: quality of life and refactoring

### PR DESCRIPTION
**Pull request by lKiseki**
# PLEASE TEST MERGE FIRST BEFORE MERGING.
# About The Pull Request
This is a hefty bundle that is mostly meant to bring some quality of life to Runner.

## Quality of Life and code changes
- **Separated Pounce keybinds**, meaning Hunter and Runner no longer share the same keybind. This should allow players a little bit more freedom in terms of preferences.
- **Added a HUD element that indicates remaining Evasion duration**. This is only visible to xenos, and it updates in real time (as in, every time the duration ticks down, the HUD element changes to reflect that with no weird delays). Admittedly, the sprites used for this one suck major dick so if anyone can provide something better I'd appreciate it.
- **Merged Toggle Savage into Pounce**. Pounce's right click / alternate use will now toggle Savage on or off. There was no real need to keep these two buttons separated, so here we are. As part of this change, I changed Pounce just a little bit and adjusted code where necessary, moving it to Hunter and making a child ability for Runner and its stuff.
  - Some Hunter and Praetorian code was tweaked accordingly to account for this refactoring, since they borrowed some code from Runner.
- **Evasion can now be stacked**. Have you ever run into a scenario where you successfully get the cooldown reset bonus, but you have to wait until Evasion is over to use it again? No more.
  - Using Evasion again while it is active will add onto Evasion's remaining duration. For example, if you currently have 1 second of Evasion and you use Evasion again, you would add 2 more seconds, extending its duration to 3 seconds.
- **Added Auto-Evasion**. This is Evasion's right click / alternate use. While Auto-Evasion is active, it will automatically attempt to use Evasion again (includes plasma costs and whatnot) if you get the cooldown reset bonus. Your life, made easier.
- **Minor code changes in Evasion**, removing old code and accounting for these changes. Also fixed a small bug where Pounce could cancel Evasion if it was currently active, and removed an easter egg that was no longer attainable in-game ever since past balance changes.
- **All aforementioned abilities that have a right click / alternate use also have an indicator on their ability icon** that displays whether they're on or off. Visual feedback, my beloved.

## Minor balance changes
- **Pounce now goes through xenos**. Other xenos will no longer block you whenever you try to pounce at something. This was moreso intended as quality of life since others blocking your pounce feels bad, but I suppose it's also a balance change.

## Other incredibly minor stuff
- So Runner's caste description was "A fast, four-legged terror, but weak in sustained combat".
  - I didn't like that "but", so I changed it to "A fast, four-legged terror. Weak in sustained combat". It looks better. It sounds better. It satisfies my OCD. Please don't question it.
- **Afterimages (as in the visual effect) will now fade away.** I am surprised they didn't do this before. They just appeared there then abruptly disappeared. Didn't look good. Now it does.
- Finishes deprecating a `hardcore` var. It was half deprecated in code, and its existence confused me, so now it's gone.

# Why It's Good For The Game
- **Runner is easier to play**. By adding some quality of life changes that make the caste easier to play in terms of execution, all players regardless of skill should have a much better time playing this caste.
  - These changes should not have an impact for more experienced players, since they can just disable things like Auto-Evasion and whatnot at will for the classic Runner experience (TM).
- **Cleaner code is always better**. Old and unnecessary code, begone.
- **Satisfies my OCD**. I am such a happy lad now.

# Changelog
:cl: Lewdcifer
add: Runners now have a new HUD element that indicates Evasion's remaining duration. This updates in real time and is visible to all xenos.
add: Added Auto-Evasion. Right click or alternate use of Evasion toggles this on and off. While Auto-Evasion is active, it will automatically try to use Evasion again if you get the cooldown reset bonus.
del: Removed Toggle Savage. It has been merged with Pounce as its right click / alternate use, otherwise retaining its usual functionality.
qol: Using Evasion again while it is active will extend its current duration instead of flipping the bird at you and making you wait until Evasion is over.
balance: Pounce will now go through xenos.
refactor: Minor code changes to Pounce. Moved it to Hunter's and gave Runner its own child ability with Toggle Savage.
code: Minor code changes to Evasion that remove unnecessary code and account for other changes.
code: Finishes deprecating a hardcore var that wasn't used anywhere.
fix: Fixed a bug where using Pounce interrupted Evasion if it was active.
imageadd: Evasion counter sprites. They suck. Please sprite something better, I beg you.
imageadd: Small indicators on ability icons for visual feedback.
spellcheck: Runner's caste description changed ever so slightly to look and sound better.
/:cl:

### CODE REVIEW

**Comment by lKiseki** [comment]
This is ready for maintainer review, it's only a draft because I may or may not replace the primordial ability.
By all means, put me on the grill and burn me alive.

**Comment by lKiseki** [comment]
Too pretentious considering it keeps the caste almost exactly as is in practice.

**Comment by FosterGrime-Heymynameslouie** [comment]
didnt even know you can stack evasion what

**Comment by lKiseki** [comment]
As of the latest commit, this PR no longer touches anything in terms of balance.

File: code/datums/keybinding/xeno.dm [hunk 1]
```diff
@@ -782,19 +782,31 @@
 	description = ""
 	keybind_signal = COMSIG_XENOABILITY_RAVAGE_SELECT
 
+/datum/keybinding/xeno/runner_pounce
+	name = "runner_pounce"
```
Comment (TiviPlus) [inline code/datums/keybinding/xeno.dm:786]: can make this "pounce" so it doesnt reset it

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 2]
```diff
@@ -1,306 +1,154 @@
...
-	to_chat(owner, span_xenodanger("We're ready to pounce again."))
-	playsound(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
-	var/mob/living/carbon/xenomorph/X = owner
-	X.usedPounce = FALSE
-	return ..()
-
-/datum/action/xeno_action/activable/pounce/use_ability(atom/A)
-	var/mob/living/carbon/xenomorph/X = owner
+	if(savage_activated)
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:28]: early return

File: code/modules/mob/living/carbon/xenomorph/castes/runner/runner.dm [hunk 3]
```diff
@@ -27,3 +27,20 @@
...
+	if(!holder)
+		return
+	holder.overlays.Cut()
+	holder.icon_state = "hudblank"
+	if(stat == DEAD || !duration)
+		return
+	holder.icon_state = "evasion_duration[duration]"
+	holder.pixel_x = 24
+	holder.pixel_y = 24
+	hud_list[XENO_EVASION_HUD] = holder
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/runner.dm:46]: cant you just add an image to client.images ior do you want other xenos seeing it too

File: code/datums/keybinding/xeno.dm (hunk 1 above)
Comment (lKiseki) [reply code/datums/keybinding/xeno.dm:786]: Sure. Donezo.

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm (hunk 2 above)
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:28]: Donezo.

File: code/modules/mob/living/carbon/xenomorph/castes/runner/runner.dm (hunk 3 above)
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/runner/runner.dm:46]: Other xenos being able to see it would be cool, yeah.

**Review by TiviPlus — APPROVED** [review]

**Comment by TheWiznard** [comment]
besides the evasion timer getting stuck at 1 or 2s on your hud after you evade occasionally I think this is pretty good as a self proclaimed runner main

**Comment by lKiseki** [comment]
> besides the evasion timer getting stuck at 1 or 2s on your hud after you evade occasionally I think this is pretty good as a self proclaimed runner main

Thank you! This bug happens when your Evasion is forcibly cancelled by something like fire or nades. It fails to update the HUD and stops updating it thereafter, so you're stuck with a number on your head until you use Evasion again. I will be fixing this on my next commit, together with the merge conflict.

**Comment by zzurge** [comment]
Only thing i dont really agree with is being able to pounce through friendlies....Otherwise good stuff

**Pull request merged**